libblkid/src/superblocks/bitlocker.c

```diff
@@ -149,26 +182,50 @@ static int probe_bitlocker(blkid_probe pr,
 {
 	const unsigned char *buf_fve = NULL;
 	const unsigned char *buf_hdr = NULL;
+	const struct bde_fve_metadata_entry *entry;
 	int rc, kind;
+	uint64_t off;
 
 	rc = get_bitlocker_headers(pr, &kind, &buf_hdr, &buf_fve);
 	if (rc)
 		return rc;
 
-	if (kind == BDE_VERSION_WIN7) {
-		const struct bde_header_win7 *hdr = (const struct bde_header_win7 *) buf_hdr;
-
-		/* Unfortunately, it seems volume_serial is always zero */
-		blkid_probe_sprintf_uuid(pr,
-				(const unsigned char *) &hdr->volume_serial,
-				sizeof(hdr->volume_serial),
-				"%016d", le32_to_cpu(hdr->volume_serial));
-	}
-
 	if (buf_fve) {
 		const struct bde_fve_metadata *fve = (const struct bde_fve_metadata *) buf_fve;
 
-		blkid_probe_sprintf_version(pr, "%d", fve->version);
+		blkid_probe_sprintf_version(pr, "%d", le16_to_cpu(fve->block_header.version));
```

Comment:
Before your change, this was broken on big-endian, correct?
If so, could you add these fixes as extra commits before the introduction of the testcase.
This way they can be backported.

Reply:
Will do!